cfme/middleware/deployment.py

```diff
@@ -43,91 +56,101 @@ class MiddlewareDeployment(MiddlewareBase):
     Args:
         name: Name of the deployment
         provider: Provider object (HawkularProvider)
+        server: Server object of the deployment (MiddlewareServer)
         id: Native id (internal id) of deployment
-        server: Server name of the deployment
 
     Usage:
 
         mydeployment = MiddlewareDeployment(name='Foo.war',
-                                server='Bar-serv',
+                                server=ser_instance,
                                 provider=haw_provider)
 
-        deployments = MiddlewareDeployment.deployments()
+        deployments = MiddlewareDeployment.deployments() [or]
+        deployments = MiddlewareDeployment.deployments(provider=haw_provider) [or]
+        deployments = MiddlewareDeployment.deployments(provider=haw_provider,server=ser_instance)
 
     """
     property_tuples = [('name', 'name')]
 
     def __init__(self, name, provider=None, **kwargs):
         if name is None:
             raise KeyError("'name' should not be 'None'")
+        if 'server' not in kwargs or not isinstance(kwargs['server'], MiddlewareServer):
+            raise KeyError("'server' should be an instance of MiddlewareServer")
         self.name = name
+        self.server = kwargs['server']
         self.provider = provider
         self.id = kwargs['id'] if 'id' in kwargs else None
-        self.server = kwargs['server'] if 'server' in kwargs else None
 
     @classmethod
-    def deployments(cls, provider=None):
+    def deployments(cls, provider=None, server=None):
         deployments = []
-        if provider:
-            # if provider instance is provided try to navigate provider's deployments page
-            # if no deployments are registered in provider, returns empty list
-            if not provider.load_all_provider_deployments():
-                return []
-        else:
-            # if provider instance is not provided then navigates  to all deployments page
-            sel.force_navigate('middleware_deployments')
+        _get_deployments_page(provider=provider, server=server)
         if sel.is_displayed(list_tbl):
-            for page in paginator.pages():
+            _server = server
+            for _ in paginator.pages():
                 for row in list_tbl.rows():
-                    deployments.append(MiddlewareDeployment(name=row.deployment_name.text,
-                                                            server=row.server.text))
+                    if not server:
+                        _server = MiddlewareServer(provider=provider, name=row.server.text)
```

Comment:
I think it is better to create server for each row, _server = MiddlewareServer(provider=provider, name=row.server.text)
and put that instance in deployment object.
To verify each deployment's displayed server value.